azkaban-web-server/src/main/java/cloudflow/servlets/ExecutionServlet.java

```diff
@@ -132,9 +122,95 @@ protected void handleGet(HttpServletRequest req, HttpServletResponse resp, Sessi
+      if (executionParameters == null) {
+        return;
+      }
+
+      String executionId;
+      try {
+        executionId = executionService.createExecution(executionParameters);
+      } catch (CloudFlowValidationException e) {
+        sendErrorResponse(resp, HttpServletResponse.SC_BAD_REQUEST, e.getMessage());
+        return;
+      } catch (CloudFlowException e) {
+        sendErrorResponse(resp, HttpServletResponse.SC_INTERNAL_SERVER_ERROR,
+            DEFAULT_500_ERROR_MESSAGE);
+        return;
+      }
+      logger.info(String.format("New execution of flow %s was successfully queued with id %s",
+          executionParameters.getFlowId(), executionId));
+      resp.setHeader("Location", GET_SINGLE_EXECUTION_URI.createURI(executionId));
+      sendResponse(resp, HttpServletResponse.SC_CREATED, new HashMap<>());
+    } else {
+      sendErrorResponse(resp, HttpServletResponse.SC_NOT_FOUND, DEFAULT_404_ERROR_MESSAGE);
+    }
+  }
+
+  private ExecutionParameters extractExecutionParameters(final HttpServletRequest req,
+    HttpServletResponse resp, Session session) throws IOException {
+    ExecutionParameters executionParameters;
+    try {
+      String body = HttpRequestUtils.getBody(req);
+      executionParameters = this.objectMapper.readValue(body, ExecutionParameters.class);
+    } catch (JsonMappingException e) {
+      final String errorMessage =
+          String.format("Parameter '%s' isn't valid.", e.getPath().get(0).getFieldName());
+      logger.error(errorMessage + " \n" + e.getMessage());
+      sendErrorResponse(resp, HttpServletResponse.SC_BAD_REQUEST, errorMessage);
+      return null;
+    } catch (JsonParseException e) {
+      final String errorMessage = "Malformed JSON content.";
+      logger.error(errorMessage + " \n" + e.getMessage());
+      sendErrorResponse(resp, HttpServletResponse.SC_BAD_REQUEST, errorMessage);
+      return null;
+    } catch (Exception e) {
+      logger.error(DEFAULT_500_ERROR_MESSAGE, e);
+      sendErrorResponse(resp, HttpServletResponse.SC_INTERNAL_SERVER_ERROR,
+          DEFAULT_500_ERROR_MESSAGE);
+      return null;
+    }
+
+    if(executionParameters.getFlowId() == null || executionParameters.getFlowId().isEmpty()) {
+      sendErrorResponse(resp, HttpServletResponse.SC_BAD_REQUEST,
+          String.format("Parameter '%s' is required.", FLOW_ID_KEY));
+      return null;
+    }
+
+    if(executionParameters.getFlowVersion() == null) {
+      sendErrorResponse(resp, HttpServletResponse.SC_BAD_REQUEST,
+          String.format("Parameter '%s' is required.", FLOW_VERSION_KEY));
+      return null;
+    } else if(executionParameters.getFlowVersion() <= 0) {
+      final String errorMessage = String.format("Parameter '%s' must be a positive number.", FLOW_VERSION_KEY);
+      logger.error(errorMessage);
+      sendErrorResponse(resp, HttpServletResponse.SC_BAD_REQUEST, errorMessage);
+      return null;
+    }
```

Comment:
We should discuss moving checks that fall under business logic to the service layer. That way the move to CF is a little easier and servlets can be removed/replaced with less changes.